GooglePlayInstant/Editor/QuickDeploy/PlayInstantSceneTreeView.cs

```diff
@@ -42,32 +44,67 @@ public PlayInstantSceneTreeView(TreeViewState treeViewState)
         public class SceneItem : TreeViewItem
         {
             public bool Enabled;
+            public bool OldEnabledValue;
+            public string SceneBuildIndexString;
+            
+            public override bool Equals(object obj)
+            {
+                var that = obj as SceneItem;
+
+                if (that == null)
+                {
+                    return false;
+                }
+
+                return this.displayName.Equals(that.displayName);
+            }
         }
 
-        protected override TreeViewItem BuildRoot()
+        public void AddOpenScenes()
         {
             var scenes = GetAllScenes();
 
-            var root = new TreeViewItem
-            {
-                id = 0,
-                depth = -1,
-                displayName = "Root"
-            };
-
-            var allItems = new List<TreeViewItem>();
             for (var i = 0; i < scenes.Length; i++)
             {
-                allItems.Add(new SceneItem
+                var sceneItem = new SceneItem
                 {
                     id = i,
                     depth = 0,
                     displayName = scenes[i].path,
                     Enabled = true
-                });
+                };
+
+                if (!_allItems.Contains(sceneItem))
+                {
+                    _allItems.Add(sceneItem);
+                }
             }
 
-            SetupParentsAndChildrenFromDepths(root, allItems);
+            EditSceneBuildIndexString();
+            Reload();
+        }
+
+        private void EditSceneBuildIndexString()
+        {
+            var i = 0;
+            foreach (var item in _allItems)
+            {
+                var sceneItem = (SceneItem) item;
+                sceneItem.id = i;
+                sceneItem.SceneBuildIndexString = sceneItem.Enabled ? "" + i++ : "";
```

Comment:
This is strange. You're only incrementing the ID if a scene is enabled, but on the previous line you're always setting the ID. So in this case with 3 scenes:
enabled
disabled
enabled

you'd end up with sceneItem.id as follows:
enabled 0
disabled 1
enabled 1

Please revisit.

Reply:
Thanks for this issue. I've reproduced this and it's exactly as you said. I don't know why I was initially changing the sceneItem id to begin with. Thanks!